### ISSUE TITLE
onSnapToItem not working (android)

### ISSUE THREAD

**Issue by sidferreira** [archriss/react-native-snap-carousel#172]
Hi! I'm using version 3.3.2 in a brand new project and the `snapToItem` isn't working. The code is exactly the one below:

```
          <Carousel
              ref={(c) => { this._carousel = c; }}
              data={pickerList}
              renderItem={({item, index}) => <Text 
                style={{
                  height: 30, 
                  textAlignVertical: 'center',
                  flex: 1, 
                  textAlign: 'center',
                  fontSize: 16
                }} key={"money"+item}>{item}</Text>}
              sliderHeight={80}
              sliderWidth={90}
              itemHeight={30}
              itemWidth={90}
              vertical
              loop
              onSnapToItem={pickerValue => console.log(pickerValue)}
            />
```

Any idea what's wrong?

**Comment by bd-arc** [comment]
Hi @sidferreira,

First, have you read [this note](https://github.com/archriss/react-native-snap-carousel#unreliable-callbacks) as well as [this one](https://github.com/archriss/react-native-snap-carousel#android-performance) from the doc?

>**TL;DR**
Android's debug mode is a mess: timeouts regularly desynchronize and scroll events are fired with some lag, which completely alters the logic of the carousel. On Android, you *will* experience issues with carousel's behavior when JS Dev Mode is enabled, and you *might* have trouble with unreliable callbacks when it isn't. This is unfortunate, but it is rooted in various flaws of the `ScrollView`/`FlatList`'s implementation and the miscellaneous workarounds we had to implement to compensate for it.
**Therefore you should always check if the issue also happens in a production build. This is also the only way to test the real performance and behavior of the carousel.**

Then, can you tell me if the issue is that `onSnapToItem` isn't fired or if it's that `pickerValue` returns a wrong value?

Finally, is [the  provided example](https://github.com/archriss/react-native-snap-carousel/tree/master/example) working properly for you?

**Comment by bd-arc** [comment]
Closing as no further feedback was provided.

**Comment by akiwarheit** [comment]
I noticed that Android's `onSnapToItem` was never reliable when I swiped too fast or when I reached the edges.

`enableMomentum` mitigated the issue - it calls `onSnapToItem` albeit with a little delay.

**Comment by bd-arc** [comment]
Hey @akiwarheit,

Have you read the following sections of the documentation?
- [Important note regarding Android](https://github.com/archriss/react-native-snap-carousel#important-note-regarding-android)
- [Android performance](https://github.com/archriss/react-native-snap-carousel/blob/master/doc/KNOWN_ISSUES.md#android-performance)
- [Unreliable callbacks](https://github.com/archriss/react-native-snap-carousel/blob/master/doc/KNOWN_ISSUES.md#unreliable-callbacks)
